language-server/src/main/java/org/sonarlint/languageserver/SonarLintLanguageServer.java

```diff
@@ -875,6 +866,18 @@ public void didChangeWorkspaceFolders(DidChangeWorkspaceFoldersParams params) {
     }
   }
 
+  @Override
+  public void setTraceNotification(SetTraceNotificationParams params) {
+    this.traceLevel = parseTraceLevel(params.getValue());
+  }
+
+  private static TraceValues parseTraceLevel(@Nullable String trace) {
+    return Optional.ofNullable(trace)
+      .map(String::toUpperCase)
+      .map(TraceValues::valueOf)
```

Comment:
This has the potential to throw exceptions when `trace` is not one of the accepted trace levels. I'd be more comfortable e.g with a "safe parse" method on `TraceValues` (with default OFF).